SanctumLabs/curtz
# Delete existing shortened URL Handler

# Description

Adds a handler to delete an existing shortened URL.

Fixes #18 

## Resolution/What did you add?

- New Handler to handle deletion of existing shortened urls
- Additional Tests
- Split up URL Repository into Read & Write interfaces

## Additional Release Steps

None

## How can we test this?

1. Run tests with `make test`
2. Run application with `docker compose up` first & then with `make run-dev` or `make run`
3. Shorten as many URLs
4. Delete a few or all of the already created URLs
5. Fetching deleted URLs should return a 404 response code

Please also note any relevant details for your test configuration.

## Checklist

- [x] Is my code backwards compatible
- [x] Have I written unit tests
